crates/ty_server/tests/e2e/code_actions.rs

```diff
@@ -309,3 +309,39 @@ html.parser
 
     Ok(())
 }
+
+/// Regression test for a panic when a code-fix diagnostic points at a string annotation
+#[test]
+fn code_action_invalid_string_annotations() -> Result<()> {
```

Comment:
Can we make this an integration test instead of an e2e test (they're so verbose and the snapshots are somewhat hard to review). See https://github.com/astral-sh/ruff/blob/258b1fd7ebca0587454c22c50b897787e03d2739/crates/ty_ide/src/code_action.rs#L86